test/box/select.result

```diff
@@ -684,3 +684,36 @@ lots_of_links
 s:drop()
 ---
 ...
+-- gh-6501 attempt to index local 'opts' (a number value)
+s = box.schema.space.create('select', { temporary = true })
+---
+...
+_ = s:create_index('primary', { type = 'tree' })
+---
+...
+s:select(0, box.index.GT)
+---
+- error: Unknown iterator type '6'
+...
+s:select(0, 'GT')
+---
+- []
+...
+_ = s:pairs(nil, box.index.GT)
+---
+- error: Unknown iterator type '6'
```

Comment:
So I can write

```lua
s:paris(nil, 'GT')
s:paris(nil, {iterator = 'GT'})
s:pairs(nil, {iterator = box.index.GT})
```

but not

```lua
s:pairs(nil, box.index.GT)
```

This is confusing. I think we should allow to pass `box.index.GT` directly. It doesn't seem to contradict the [documentation](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_index/pairs/#lua-function.index_object.pairs).

Reply:
Fixed it.